fe/fe-core/src/main/java/com/starrocks/task/ExportExportingTask.java

```diff
@@ -339,18 +341,29 @@ protected void exec() {
 
                 if (i < RETRY_NUM - 1) {
                     TUniqueId queryId = coord.getQueryId();
-                    coord.clearExportStatus();
-
-                    // generate one new queryId here, to avoid being rejected by BE,
-                    // because the request is considered as a repeat request.
-                    // we make the high part of query id unchanged to facilitate tracing problem by log.
                     UUID uuid = UUID.randomUUID();
-                    TUniqueId newQueryId = new TUniqueId(queryId.hi, uuid.getLeastSignificantBits());
-                    coord.setQueryId(newQueryId);
+                    if (exportJob.needResetCoord()) {
+                        try {
+                            Coordinator newCoord = exportJob.resetCoord(taskIdx, uuid);
+                            coord = newCoord;
+                        } catch (UserException e) {
+                            LOG.warn("fail to reset coord for task idx: {}, task query id: {}, reason: {}", taskIdx, 
+                                    getQueryId(), e.getMessage());
+                            // still use old coord if there are any problems when reseting Coord
+                            coord.clearExportStatus();
+                            coord.setQueryId(new TUniqueId(queryId.hi, uuid.getLeastSignificantBits()));
+                        }
+                    } else {
+                        coord.clearExportStatus();
+                        // generate one new queryId here, to avoid being rejected by BE,
+                        // because the request is considered as a repeat request.
+                        // we make the high part of query id unchanged to facilitate tracing problem by log.
+                        coord.setQueryId(new TUniqueId(queryId.hi, uuid.getLeastSignificantBits()));
+                    }
                     LOG.warn(
                             "export sub task fail. err: {}. task idx: {}, task query id: {}. retry: {}, new query id: {}",
                             coord.getExecStatus().getErrorMsg(), taskIdx, DebugUtil.printId(queryId), i,
-                            DebugUtil.printId(newQueryId));
+                            DebugUtil.printId(coord.getQueryId()));
```

Comment:
When retry the export task, does the queryId changed or not?

Reply:
changed,
if queryId is not changed, be will think the planfragment is same as before, and refuse to execute it.